src/Servers/Kestrel/Core/src/Internal/AddressBinder.cs

```diff
@@ -141,6 +141,31 @@ internal static ListenOptions ParseAddress(string address, out bool https)
         }
 
         return options;
+
+        bool IsLocalhost(string host, out string? prefix)
+        {
+            // Check for "localhost"
+            if (string.Equals(parsedAddress.Host, "localhost", StringComparison.OrdinalIgnoreCase))
+            {
+                prefix = null;
+                return true;
+            }
+
+            // Check for use of .localhost TLD (Top Level Domain)
+            if (host.EndsWith(".localhost", StringComparison.OrdinalIgnoreCase))
+            {
+                var localhostTldIndex = host.LastIndexOf('.');
```

Comment:
Actually, we already checked for `.localhost`, so we can just do `host[..^10]`? 😆